consoleme/lib/account_indexers/aws_organizations.py

```diff
@@ -59,3 +70,235 @@ async def retrieve_accounts_from_aws_organizations() -> CloudAccountModelArray:
+    **kwargs
+) -> List[Dict[str, Any]]:
+    """Wrapper for organizations:ListChildren
+
+    Args:
+        ca: CloudAux instance
+        parent_id: ID of organization root or organizational unit
+        child_type: ACCOUNT or ORGANIZATIONAL_UNIT
+    """
+    return ca.call(
+        "organizations.client.list_children",
+        ChildType=child_type,
+        ParentId=parent_id,
+        **kwargs
+    )
+
+
+@paginated(
+    "Roots",
+    response_pagination_marker="NextToken",
+    request_pagination_marker="NextToken",
+)
+def _list_org_roots(ca: CloudAux, **kwargs) -> List[Dict[str, Any]]:
+    """Wrapper for organizations:ListRoots
+
+    Args:
+        ca: CloudAux instance
+    """
+    return ca.call("organizations.client.list_roots", **kwargs)
+
+
+def _get_children_for_ou(ca: CloudAux, root_id: str) -> Dict[str, Any]:
+    """Recursively build OU structure
+
+    Args:
+        ca: CloudAux instance
+        root_id: ID of organization root or organizational unit
+    """
+    children: List[Dict[str, Any]] = []
+    children.extend(_list_children_for_ou(ca, root_id, "ORGANIZATIONAL_UNIT"))
+    children.extend(_list_children_for_ou(ca, root_id, "ACCOUNT"))
+    for child in children:
+        child["Parent"] = root_id
+        if child["Type"] == "ORGANIZATIONAL_UNIT":
+            child.update(_describe_ou(ca, child["Id"]))
+            child["Children"] = _get_children_for_ou(ca, child["Id"])
+        else:
+            child.update(_describe_account(ca, child["Id"]))
+    return children
+
+
+async def retrieve_org_structure(
+    org_account_id: str, region: str = "us-east-1"
+) -> Dict[str, Any]:
+    """Retrieve org roots then recursively build a dict of child OUs and accounts.
+
+    This is a slow and expensive operation.
+
+    Args:
+        org_account_id: ID for AWS account containing org(s)
+        region: AWS region
+    """
+    conn_details = {
+        "assume_role": config.get("policies.role_name"),
+        "account_number": org_account_id,
+        "session_name": "ConsoleMeSCPSync",
+        "region": region,
+    }
+    ca = CloudAux(**conn_details)
+    roots = _list_org_roots(ca)
+    org_structure = {}
+    for root in roots:
+        root_id = root["Id"]
+        root["Children"] = _get_children_for_ou(ca, root["Id"])
+        org_structure[root_id] = root
+    return org_structure
+
+
+async def retrieve_scps_for_organization(
+    org_account_id: str, region: str = "us-east-1"
+) -> List[ServiceControlPolicyModel]:
+    """Return a ServiceControlPolicyArrayModel containing all SCPs for an organization
+
+    Args:
+        org_account_id: ID for AWS account containing org(s)
+        region: AWS region
+    """
+    conn_details = {
+        "assume_role": config.get("policies.role_name"),
+        "account_number": org_account_id,
+        "session_name": "ConsoleMeSCPSync",
+        "region": region,
+    }
+    ca = CloudAux(**conn_details)
+    all_scp_metadata = _list_service_control_policies(ca)
```

Comment:
We should wrap this call in a threadpool, ie: `await sync_to_async(_list_service_control_policies)(ca)`

Reply:
Might make sense to migrate the _list_service_control_policies function to cloudaux at some point